owlkettle/adw.nim

```diff
@@ -563,6 +565,119 @@ proc `valSwipe=`*(flap: Flap, swipe: bool) =
   flap.valSwipeToOpen = swipe
   flap.valSwipeToClose = swipe
 
+type PackType* = enum
+  PackStart, PackEnd
+
+type WindowControlButton* = enum
+  WindowControlMinimize = "minimize"
+  WindowControlMaximize = "maximize"
+  WindowControlClose = "close"
+  WindowControlIcon = "icon"
+  WindowControlMenu = "menu"
+
+proc toLayoutString(buttons: seq[WindowControlButton]): string =
+  return buttons.mapIt($it).join(",")
+
+renderable AdwHeaderBar of BaseWidget:
```

Comment:
We need a consistent way of dealing with name collisions. I think the ideal solution would be to support qualified names in `guidsl`. This is a reminder to myself to implement that.

Reply:
Honestly I like the Adw Prefix simply because it informs me where the widget comes from.
It also tells me that if I have the chance to choose between 2 widget versions, I should use the adwaita one since its likely to have more features.

Reply:
I added support for qualified names, however after prototyping a bit, I think I agree with you that a prefix is the better solution here.

Reply:
For consistencies sake then the following question:
I assume that we want to use the Adw prefix for every widget where we have ambiguous naming (or maybe add the prefix for all widgets in adw?).
Should we rename WindowSurface to AdwWindow for naming consistency, given that that's a breaking change and we have 3.0 just around the corner?

Reply:
It would make sense, yes.